### ISSUE TITLE
STUN/TURN blocking exceptions

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/CoreLibs#1567]
There're several apps that use STUN/TURN for legitimate purposes.

For instance, WhatsApp:
https://github.com/AdguardTeam/AdguardForAndroid/issues/3627#issuecomment-991664862

We should be able to make exceptions for those cases, the question is what should be the way?
`$stealth` exception rules seem to be a good fit, but to make them work properly, we need to get the target hostname.

Relevant issue: https://github.com/AdguardTeam/CoreLibs/issues/66

**Comment by TPS** [comment]
Google Voice Wifi calling is also blocked due to this.

**Comment by TPS** [comment]
I couldn't find any toggle to disable `stun` blocking outright in AGA.  Does such exist anyplace?

**Comment by TPS** [comment]
> I couldn't find any toggle to disable `stun` blocking outright in AGA.  Does such exist anyplace?

Is it handled by the `Block WebRTC` toggle in the Stealth module?  Or is that covering more/other things?

**Comment by ameshkov** [comment]
> Is it handled by the Block WebRTC toggle in the Stealth module? Or is that covering more/other things?

Yep, it is.

**Comment by TPS** [comment]
Does that toggle have any _other_ effects?  Or is purely for wholesale `STUN`/`TURN` blocking?  Will these still show up in the AG log (but just not blocked) for me to track how often these appear generally?

**Comment by ngorskikh** [comment]
@ameshkov The issue with WhatsApp etc. is actually a recent regression introduced when AdGuard for Android started redirecting UDP traffic to CoreLibs. AdGuard for Android never blocked STUN/TURN traffic for non-browser applications, but CoreLibs doesn't make that exception (neither do our applications for any other platform, AFAICT). Maybe a hotfix for AdGuard for Android is in order, i.e. don't redirect UDP traffic to CoreLibs in the above exception case.

**Comment by ameshkov** [comment]
@ngorskikh I'll file a separate issue about it to Android and we'll try to come up with a solution asap.

Meanwhile, I still think we need a way to control STUN/TURN blocking by the filtering rules (consider the case when you want to unblock it for web.whatsapp.com for instance).

**Comment by TPS** [comment]
Is there a way to generalize all these issues so that _other_ communications apps/sites (e.g., GVoice) are not forgotten?

**Comment by ameshkov** [comment]
My original point was that we should grant filtering rules control over that stealth mode feature, this is as generalized as it gets :)

**Comment by TPS** [comment]
Even w/ the `Block WebRTC` toggle off, GVoice calls don't complete (they ring, but hangup as soon as answered with "Network blocked" error), but will when AG is disabled outright.  Maybe resolving _just_ this issue doesn't seem to be enough to fix the actual problem?
